modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/ShelfBottomSheet.kt

```diff
@@ -79,16 +96,28 @@ class ShelfBottomSheet : BaseDialogFragment() {
                     (activity as? FragmentHostListener)?.openCloudFiles()
                 }
             }
-            is ShelfItem.Cast -> { binding?.mediaRouteButton?.performClick() }
+            is ShelfItem.Cast -> {
+                analyticsAction = ShelfItem.ShelfItemId.CAST.analyticsValue
+                binding?.mediaRouteButton?.performClick()
+            }
             is ShelfItem.Played -> {
+                analyticsAction = ShelfItem.ShelfItemId.PLAYED.analyticsValue
                 context?.let {
                     playerViewModel.markCurrentlyPlayingAsPlayed(it)?.show(parentFragmentManager, "mark_as_played")
                 }
             }
-            is ShelfItem.Archive -> { playerViewModel.archiveCurrentlyPlaying(resources)?.show(parentFragmentManager, "archive") }
-            else -> {}
+            is ShelfItem.Archive -> {
+                analyticsAction = ShelfItem.ShelfItemId.ARCHIVE.analyticsValue
+                playerViewModel.archiveCurrentlyPlaying(resources)?.show(parentFragmentManager, "archive")
+            }
+            else -> {
+                analyticsAction = AnalyticsProp.Value.UNKNOWN
+            }
```

Comment:
I think the only missing branch here is for `Download`, so what if Instead of having an `else` branch here we added that branch specifically. I feel like that would be a bit more informative about which items we're not expecting here and if we add a new `ShelfItem` in the future, we would get a compile error here and force us to explicitly decide how we want to handle the new item instead of it just getting caught in the else. wdyt?

(note that the commit I linked to in my other comment includes this change)

Reply:
Makes sense to me 👍
Accepted in b5c5a8e